core/templates/dev/head/pages/topic-viewer-page/subtopics-list/subtopics-list.directive.ts

```diff
@@ -0,0 +1,140 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the subtopics list.
+ */
+
+require('components/summary-tile/subtopic-summary-tile.directive.ts');
+
+require('domain/utilities/UrlInterpolationService.ts');
+require('services/contextual/WindowDimensionsService.ts');
+
+angular.module('oppia').directive('subtopicsList', [
+  'UrlInterpolationService', function(UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {
+        getSubtopics: '&subtopicsList',
+      },
+      templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+        '/pages/topic-viewer-page/subtopics-list/' +
+        'subtopics-list.directive.html'),
+      controller: [
+        'WindowDimensionsService', '$scope', '$timeout', 'UrlService',
+        function(
+            WindowDimensionsService, $scope, $timeout, UrlService) {
+          var SUBTOPIC_TILE_WIDTH_PX = 310;
+          $scope.leftmostCardIndices = 0;
+          var MAX_NUM_TILES_PER_ROW = 4;
+          $scope.tileDisplayCount = 0;
+          $scope.topicName = UrlService.getTopicNameFromLearnerUrl();
+          var initCarousels = function() {
+            $scope.subtopics = $scope.getSubtopics();
+            if (!$scope.subtopics) {
+              return;
+            }
+
+            var windowWidth = $(window).width();
```

Comment:
Don't just copy code wholesale from the library page -- pull out common code into a common directive and reuse it here. (Remember the DRY principle.)

Reply:
Created issue #7358.